### DISCUSSION TITLE
Browser Play via TeaVM

### DISCUSSION THREAD

**Discussion by xechor** [MinicraftPlus/minicraft-plus-revived#692] [Ideas]
Would TeaVM be viable for this? I toyed a bit with CheerpJ but the speed was abhorrent. So glad to see this project exists!

**Comment by Litorom** [comment]
I personally never heard of teaVM, is it a browser virtual machine?

**Reply by xechor** [reply]
[TeaVM](https://teavm.org/) is an ahead-of-time compiler for Java bytecode that emits JavaScript and WebAssembly that runs in a browser, not unlike GWT.

**Comment by BenCheung0422** [comment]
From my research, it converts Java source code into JavaScript and WebAssembly. According to [this answer](https://stackoverflow.com/a/53960808), it is definitely not realizable on this project, since there is extremely high possibility that we will move to using LWJGL with native features (including filesystems, OpenGL and OpenAL).

**Reply by xechor** [reply]
Good info, thank you! Possibly CheerpJ is the only potential avenue for that then similar to [Browsercraft](https://github.com/leaningtech/browsercraft).
